File: src/fiefapp.svelte/src/models/industries/felling.ts
import { Industry } from './industry';

interface IFelling {
  amountLandclearing: number;
  amountLandclearingOfFelling: number;
  amountFelling: number;
  amountClearUseless: number;
}

export class Felling extends Industry implements IFelling {
  amountLandclearing: number;
  amountLandclearingOfFelling: number;
  amountFelling: number;
  amountClearUseless: number;

  constructor() {
    super();
    this.name = 'Skogsavverkning';
    this.needSteward = true;
    this.amountLandclearing = 0;
    this.amountLandclearingOfFelling = 0;
    this.amountFelling = 0;
    this.amountClearUseless = 0;
    this.wood = 0;
  }
}

File: src/fiefapp.entities/Felling.cs
using fiefapp.entities.BaseClasses;
using fiefapp.entities.Interfaces;
using System;

namespace fiefapp.entities
{
    public class Felling : Industry, IBaseEntity, INeedSteward, ICanBeDeveloped, IFiefForeinKey
    {
        public Felling()
        {
            Id = Guid.NewGuid();
            Name = "Skogsavverkning";
            IsInformationItem = true;
        }

        public override Guid Id { get; set; }
        public Guid FiefId { get; set; }
        public override string Name { get; set; }
        public override int? Silver { get; set; }
        public override int? Base { get; set; }
        public override int? Luxury { get; set; }
        public override int? Wood { get; set; }
        public override int? Stone { get; set; }
        public override int? Iron { get; set; }
        public int AmountLandclearing { get; set; }
        public int AmountLandclearingOfFelling { get; set; }
        public int AmountFelling { get; set; }
        public int AmountClearUseless { get; set; }
        public override int DevelopmentLevel { get; set; }
        public override bool IsBeingDeveloped { get; set; }
        public override int? Quality { get; set; }
        public override bool IsInformationItem { get; set; }
        public override Guid? DevelopmentId { get; set; }
        public override Development Development { get; set; }
        public override Guid? StewardId { get; set; }
        public override Steward Steward { get; set; }
    }
}
